### PR TITLE
Fix AWS CloudWatch start and shutdown logic

### PR DESCRIPTION
**Pull request by errordeveloper**
**Description:** Ensure that Start and Shutdown handlers are wired up correctly.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4683

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]

File: exporter/awscloudwatchlogsexporter/factory.go [hunk 1]
```diff
@@ -55,6 +55,8 @@ func createLogsExporter(_ context.Context, set component.ExporterCreateSettings,
 			Enabled:      true,
 			NumConsumers: 1, // due to the sequence token, there can be only one request in flight
 		}),
+		exporterhelper.WithStart(exporter.Start),
+		exporterhelper.WithShutdown(exporter.Shutdown),
```
Comment (bogdandrutu) [inline exporter/awscloudwatchlogsexporter/factory.go:59]: Very nit: move this as first 2 options :)

Comment (errordeveloper) [reply exporter/awscloudwatchlogsexporter/factory.go]: Sure, done.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by errordeveloper** [comment]
@bogdandrutu @Aneurysm9 thanks for speedy reviews!

**Pull request merged**